### PR TITLE
Stop cm-janus process when any module fails to start

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@vogdb please review

**Comment by tomaszdurka** [comment]
Btw this kind of process exit should happen on any unhandled rejection imo. Let's not introduce it yet though as it might kill app in some cases we don't expect yet.

File: lib/index.js [hunk 1]
```diff
@@ -67,14 +67,18 @@ Application.prototype.registerServices = function() {
...
   jobManager.handlerRegistry.registerFromConfiguration(this.config['jobManager']['handlersConfiguration']);
-  jobManager.start();
+
+  new Promise.all([
+    proxy.start(),
+    httpServer.start(),
+    jobManager.start()
+  ]).catch(function(error) {
+    serviceLocator.get('logger').error('Process failed. Exiting. ' + error.stack || error.message);
+    process.exit(1);
```
Comment (tomaszdurka) [inline lib/index.js]: @vogdb should we rather emit close here?

**Comment by vogdb** [comment]
ok but jobManager.start does not return Promise

**Comment by vogdb** [comment]
Still your solution will work

**Comment by vogdb** [comment]
lgtm

File: lib/index.js (hunk 1 above)
Comment (vogdb) [inline lib/index.js]: We can but this error happens on start so does not matter much. If you want to be a super correct then you can emit `close` instead.

**Comment by vogdb** [comment]
lgtm

**Comment by tomaszdurka** [comment]
Paired with @vogdb

**Pull request merged**